docs/getting-started/publish-infrastructure.md

```diff
@@ -0,0 +1,567 @@
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Composition
+metadata:
+  name: mysqlinstances.gcp.database.example.org
+  labels:
+    provider: gcp
+spec:
+  writeConnectionSecretsToNamespace: crossplane-system
+  reclaimPolicy: Delete
+  from:
+    apiVersion: database.example.org/v1alpha1
+    kind: MySQLInstance
+  to:
+    - base:
+        apiVersion: database.gcp.crossplane.io/v1beta1
+        kind: CloudSQLInstance
+        spec:
+          forProvider:
+            databaseVersion: MYSQL_5_7
+            region: us-central1
+            ipConfiguration:
+              ipv4Enabled: true
+              authorizedNetworks:
+                - value: "0.0.0.0/0"
+            settings:
+              tier: db-n1-standard-1
+              dataDiskType: PD_SSD
+              dataDiskSizeGb: 10
+          writeConnectionSecretToRef:
+            namespace: crossplane-system
+          providerRef:
+            name: gcp-provider
+          reclaimPolicy: Delete
+      patches:
+        - fromFieldPath: "metadata.uid"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-mysql"
+        - fromFieldPath: "spec.storageGB"
+          toFieldPath: "spec.forProvider.settings.dataDiskSizeGb"
+      connectionDetails:
+        - fromConnectionSecretKey: username
+        - fromConnectionSecretKey: password
+        - fromConnectionSecretKey: endpoint
+```
+</div>
+<div class="tab-pane fade" id="azure-tab-1" markdown="1">
+> Note: the `Composition` for Azure also includes a `ResourceGroup` and
+> `MySQLServerFirewallRule` that are required to provision a publicly available
+> MySQL instance on Azure. Composition enables scenarios such as this, as well
+> as far more complex ones. See the [composition] documentation for more
+> information.
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Composition
+metadata:
+  name: mysqlinstances.azure.database.example.org
+  labels:
+    provider: azure
+spec:
+  writeConnectionSecretsToNamespace: crossplane-system
+  reclaimPolicy: Delete
+  from:
+    apiVersion: database.example.org/v1alpha1
+    kind: MySQLInstance
+  to:
+    - base:
+        apiVersion: azure.crossplane.io/v1alpha3
+        kind: ResourceGroup
+        spec:
+          reclaimPolicy: Delete
+          providerRef:
+            name: azure-provider
+    - base:
+        apiVersion: database.azure.crossplane.io/v1beta1
+        kind: MySQLServer
+        spec:
+          forProvider:
+            administratorLogin: myadmin
+            resourceGroupNameSelector:
+              matchControllerRef: true
+            location: West US 2
+            sslEnforcement: Disabled
+            version: "5.7"
+            sku:
+              tier: GeneralPurpose
+              capacity: 2
+              family: Gen5
+            storageProfile:
+              storageMB: 20480
```

Comment:
Typically when a field like `storageMB` is required by the definition I prefer to omit it from the base, since it will always be overwritten by the value of the composite/requirement.